src/neo-vm/Types/Boolean.cs

```diff
@@ -31,7 +31,7 @@ public override bool Equals(StackItem other)
             {
                 return false;
             }
-            return Unsafe.MemoryEquals(GetByteArray(), bytes_other);
+            return Unsafe.MemoryEquals(GetByteArray().Span, bytes_other.Span);
```

Comment:
you shouldn't need an unsafe method to iterate a span

Reply:
This unsafe implementation is faster.